src/app/components/PromotionModal/index.tsx

```diff
@@ -14,7 +14,7 @@ import promotionModalStyles from './index.module.scss';
 
 const SESSION_KEY = 'sovryn-promo-dialog-1'; //increment this whenever new content is added, to ensure users see it even if they viewed one previously
 
-const shouldModalBeVisible = () => !local.getItem(SESSION_KEY);
+const shouldModalBeVisible = () => !!local.getItem(SESSION_KEY);
```

Comment:
Why did you change this? Now you basically say that the modal should be visible only if you have already seen it so it means you won't ever see it.